src/GameState/moveFunctions.tsx

```diff
@@ -22,20 +22,32 @@ export const isMovePossible = async (
   }
 };
 
-export const moveMarble = async (moveMarbleReference: MoveMarbleReference) => {
-  return await fetch(
-    `${API_URL}/games/${moveMarbleReference.gameId}/move-marble`,
+export const moveMarble = async (
+  moveMarbleReference: MoveMarbleReference | undefined,
+) => {
+  const response = await fetch(
+    `${API_URL}/games/${moveMarbleReference?.gameId}/move-marble`,
     {
       method: 'POST',
       body: JSON.stringify({
-        coordinates: moveMarbleReference.coordinates,
-        direction: moveMarbleReference.direction,
-        player: moveMarbleReference.playerForAPI,
+        coordinates: moveMarbleReference?.coordinates,
+        direction: moveMarbleReference?.direction,
+        player: moveMarbleReference?.playerForAPI,
       }),
       headers: {
         'Content-Type': 'application/json',
-        Authorization: 'Bearer ' + moveMarbleReference.playerToken,
+        Authorization: 'Bearer ' + moveMarbleReference?.playerToken,
       },
     },
   );
+
+  if (!response.ok) {
+    throw Error(response.statusText);
+  }
+
+  if (response.status >= 200 && response.status < 300) {
```

Comment:
> The ok read-only property of the Response interface contains a Boolean stating whether the response was successful (status in the range 200-299) or not.

So the conditions are useless because line 44 already checks it.